presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueryFramework.java

```diff
@@ -68,25 +68,43 @@
 
 public abstract class AbstractTestQueryFramework
 {
-    private QueryRunnerSupplier queryRunnerSupplier;
+    @Deprecated
+    private Optional<QueryRunnerSupplier> queryRunnerSupplier;
     private QueryRunner queryRunner;
     private H2QueryRunner h2QueryRunner;
     private SqlParser sqlParser;
 
+    @Deprecated
```

Comment:
I don't think it should be deprecated. I guess there might be still cases where injecting supplier is more natural choice than inheritance. However, in case of testng it sounds like an inheritance is usual the only way to go... ; /

Reply:
Injecting supplier is convenient most of the times (this is why we had it), but overridable method is more flexible.
I don't want to have 2 different ways in the long term, so I am going to completely remove "convenient but too inflexible" approach we have today.

do you agree?

Reply:
> overridable method is more flexible.

I don't think it is more flexible. Here it could be more convenient. I don't think this refactoring is needed to fix the main issue here.

Anyway, I am fine with this method.

Reply:
> I don't think this refactoring is needed to fix the main issue here.

how?

Reply:
Maybe I am missing something, but I guess you could always pass query runner supplier.

Reply:
This is what we do today. And we end up allocating resource in test constructor (and writing two constructors to capture the resource on the field).
This could be worked around by adding `onClose` to QueryRunner / ClosingQueryRunner. But that's just fixing the wrong design -- composition is nice, but mixing composition _and_ inheritance within single class hierarchy is not good.